### PR TITLE
Use listen in listenOnce

### LINKED ISSUE #7591: Use listen in listenOnce
We can reduce a bit of duplication by calling [`listen`](https://github.com/ampproject/amphtml/blob/6a97771/src/event-helper.js#L46-L68) from [`listenOnce`](https://github.com/ampproject/amphtml/blob/6a97771/src/event-helper.js#L80-L105).

### PR DESCRIPTION
**Pull request by jridgewell**
Fixes #7591.

### CODE REVIEW

File: src/event-helper.js [hunk 1]
```diff
@@ -64,30 +64,14 @@ export function listen(element, eventType, listener, opt_capture) {
...
+  let unlisten = internalListenImplementation(element, eventType, event => {
     try {
-      localListener(event);
-    } catch (e) {
-      // reportError is installed globally per window in the entry point.
-      self.reportError(e);
-      throw e;
+      listener.call(this, event);
     } finally {
+      listener = null;
```
Comment (alanorozco) [inline src/event-helper.js]: What's this assignment for?

Comment (jridgewell) [inline src/event-helper.js]: To ensure listener is released. It's a defense-in-depth like what was here before.

Comment (alanorozco) [inline src/event-helper.js]: Mind adding a short comment (`// for GC` or whatever) explaining that? I could easily see someone removing it to clean things up.

Comment (jridgewell) [inline src/event-helper.js]: Done

**Pull request merged**